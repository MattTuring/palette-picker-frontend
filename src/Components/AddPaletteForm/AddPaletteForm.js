import React from 'react';

const AddPaletteForm = (props) => {
  const changeInput = (event) => {
    props.setPaletteName(event.target.value);
  };

  const chooseProject = (event) => {
    props.setCurrentProject(+event.target.value);
  };

  const submitPalette = async () => {
    if(props.paletteName.length) {
      props.addPalette();
    }
  };

  const options = props.projects.map(project => (
    <option value={project.id} key={project.id + project.id + Math.random}>{project.name}</option>
  ));

  return (
    <section className="palettes">
      <div>
        <button className="new-color" onClick={props.assignToRandomColor}>New Colors</button>
        <input
          className="palette-name"
          onChange={changeInput}
          placeholder="Name"
          type="text"/>
        <select onChange={chooseProject}>
          {(props.projects.length) && options }
        </select>
        <button className="submit-btn" onClick={submitPalette}>Add Palette</button>
      </div>
    </section>
  )
};

export default AddPaletteForm;
