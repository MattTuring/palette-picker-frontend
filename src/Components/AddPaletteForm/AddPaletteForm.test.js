import React from 'react';
import { shallow } from 'enzyme';
import AddPaletteForm from './AddPaletteForm';

describe('AddPaletteForm', () => {
  const assignToRandomColor = jest.fn();
  const setPaletteName = jest.fn();
  const setCurrentProject = jest.fn();
  const addPalette = jest.fn();

  const mockProps = {
    colors: {
      color1: '#000000',
      color2: '#ffffff',
      color3: '#000000',
      color4: '#ffffff',
      color5: '#000000'
    },
    paletteName: 're',
    projects: [{
      id: 1,
      name: 'Project'
    }],
    assignToRandomColor,
    setPaletteName,
    setCurrentProject,
    addPalette
  };
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <AddPaletteForm {...mockProps} />);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call assignToRandomColor after click on ', () => {
    wrapper.find('.new-color').simulate('click');
    expect(assignToRandomColor).toHaveBeenCalled();
  });

  it('should call addPalette after click on submit button', () => {
    wrapper.find('.submit-btn').simulate('click');
    expect(addPalette).toHaveBeenCalled();
  });

  it('should call setPaletteName after changes in input', () => {
    wrapper.find('input').simulate('change', {
      target: {value: 'new'}
    });

    expect(setPaletteName).toHaveBeenCalled();
  });

  it('should call setCurrentProject after changes in dropmenu', () => {
    wrapper.find('select').simulate('change', {
      target: {value: 1}
    });

    expect(setCurrentProject).toHaveBeenCalledWith(1);
  });
});
