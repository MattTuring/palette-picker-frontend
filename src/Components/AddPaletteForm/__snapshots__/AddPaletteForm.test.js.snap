// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPaletteForm should match the snapshot with all data passed in correctly 1`] = `
<section
  className="palettes"
>
  <div>
    <button
      className="new-color"
      onClick={[MockFunction]}
    >
      New Colors
    </button>
    <input
      className="palette-name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
    />
    <select
      onChange={[Function]}
    >
      <option
        key="2function random() { [native code] }"
        value={1}
      >
        Project
      </option>
    </select>
    <button
      className="submit-btn"
      onClick={[Function]}
    >
      Add Palette
    </button>
  </div>
</section>
`;
